plugin/trino-paimon/src/main/java/io/trino/plugin/paimon/PaimonMetadata.java

```diff
@@ -0,0 +1,457 @@
+        }
+
+        PaimonTableOptionUtils.buildOptions(builder, properties);
+
+        return builder.build();
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        PaimonTableHandle table = (PaimonTableHandle) tableHandle;
+        try {
+            catalog.dropTable(session, new Identifier(table.getSchemaName(), table.getTableName()), false);
+        }
+        catch (TableNotExistException e) {
+            throw new TrinoException(TABLE_NOT_FOUND, "Table '%s' does not exist".formatted(table.getTableName()));
+        }
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        PaimonTableHandle table = (PaimonTableHandle) tableHandle;
+        Map<String, ColumnHandle> columnHandles = new HashMap<>();
+        for (ColumnMetadata column : table.columnMetadatas(session, catalog)) {
+            columnHandles.put(column.getName(), table.columnHandle(session, catalog, column.getName()));
+        }
+        return columnHandles;
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        PaimonColumnHandle column = (PaimonColumnHandle) columnHandle;
+        return new ColumnMetadata(column.columnName(), column.trinoType());
+    }
+
+    // TODO Implement streamRelationColumns method
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+
+        List<SchemaTableName> tableNames;
+        if (prefix.getTable().isPresent()) {
+            tableNames = Collections.singletonList(prefix.toSchemaTableName());
+        }
+        else {
+            tableNames = listTables(session, prefix.getSchema());
+        }
+
+        return tableNames.stream()
+                .map(table -> {
+                    PaimonTableHandle handle;
+                    List<ColumnMetadata> columnMetadata;
+                    try {
+                        handle = getTableHandle(session, table, Collections.emptyMap());
+                        columnMetadata = handle.columnMetadatas(session, catalog);
+                    }
+                    catch (RuntimeException e) {
+                        // Error when getting column metadata, return null
+                        return Pair.of(table, (List<ColumnMetadata>) null);
+                    }
+
+                    return Pair.of(table, columnMetadata);
+                }).filter(p -> p.getRight() != null)
+                .collect(toMap(Pair::getLeft, Pair::getRight));
+    }
+
+    @Override
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle handle, Constraint constraint)
```

Comment:
I also suggest removing applyFilter, applyProjection and applyLimit implementations from this PR. 
They can be handled as a follow-up.